src/extension.ts

```diff
@@ -84,6 +109,114 @@ function previewSelection() {
+ */
+async function startScene() {
+	const editor = vscode.window.activeTextEditor;
+	if (!editor) {
+		vscode.window.showErrorMessage(
+			'No opened file found. Please place your cursor at a line of code.'
+		);
+		return;
+	}
+
+	// Save active file
+	vscode.commands.executeCommand('workbench.action.files.save');
+
+	const languageId = editor.document.languageId;
+	if (languageId !== 'python') {
+		vscode.window.showErrorMessage("You don't have a Python file open.");
+		return;
+	}
+
+	const lines = editor.document.getText().split("\n");
+
+	// Find which lines define classes
+	// E.g. here, classLines = [{ line: "class FirstScene(Scene):", index: 3 }, ...]
+	const classLines = lines
+		.map((line, index) => ({ line, index }))
+		.filter(({ line }) => /^class (.+?)\((.+?)\):/.test(line));
+
+	const cursorLine = editor.selection.start.line;
+
+	// Find the first class defined before where the cursor is
+	// E.g. here, matchingClass = { line: "class SelectedScene(Scene):", index: 42 }
+	const matchingClass = classLines
+		.reverse()
+		.find(({ index }) => index <= cursorLine);
+	if (!matchingClass) {
+		vscode.window.showErrorMessage('Place your cursor in Manim code inside a class.');
+		return;
+	}
+	// E.g. here, sceneName = "SelectedScene"
+	const sceneName = matchingClass.line.slice("class ".length, matchingClass.line.indexOf("("));
+
+	// While line is empty - make it the previous line
+	// (because `manimgl -se <lineNumber>` doesn't work on empty lines)
+	let lineNumber = cursorLine;
+	while (lines[lineNumber].trim() === "") {
+		lineNumber--;
+	}
+
+	// Create the command
+	const filePath = editor.document.fileName;  // absolute path
+	const cmds = ["manimgl", filePath, sceneName];
+	let enter = false;
+	if (cursorLine !== matchingClass.index) {
+		cmds.push(`-se ${lineNumber + 1}`);
+		enter = true;
+	}
+	const command = cmds.join(" ");
+
+	// // Commented out - in case someone would like it.
+	// // For us - we want to NOT overwrite our clipboard.
+	// // If one wants to run it in a different terminal,
+	// // it's often to write to a file
+	// await vscode.env.clipboard.writeText(command + " --prerun --finder -w");
+
+	// Run the command
+	executeTerminalCommand(command);
+
+	// // Commented out - in case someone would like it.
+	// // For us - it would require MacOS. Also - the effect is not desired.
+	// // Focus some windows (ONLY for MacOS because it uses `osascript`!)
+	// const terminal = vscode.window.activeTerminal || vscode.window.createTerminal();
+	// if (enter) {
+	// 	// Keep cursor where it started (in VSCode)
+	// 	const cmd_focus_vscode = 'osascript -e "tell application \\"Visual Studio Code\\" to activate"';
+	// 	// Execute the command in the shell after a delay (to give the animation window enough time to open)
+	// 	await new Promise(resolve => setTimeout(resolve, 2500));
+	// 	require('child_process').exec(cmd_focus_vscode);
+	// } else {
+	// 	terminal.show();
+	// }
```

Comment:
Yes, it might be nice to leave this in for other people. But likely if we change the code here, we won't uncomment this just to make sure this still works. So this will age very quickly compared to our code. That's why I'd rather vote to get rid of it entirely, even if at this time it might be useful for some.

Other options (that I find more reasonable) would be:
- Activate this behavior based on a user setting (e.g. checkbox)
- Write this up in a GitHub issue with codeboxes, then close the issue but link to it somewhere in the documentation.